cmd/user-ssh-keys-agent/Dockerfile

```diff
@@ -14,13 +14,21 @@
 
 FROM docker.io/golang:1.16.4 as builder
 
+ARG HOST_GOCACHE
 WORKDIR /go/src/github.com/kubermatic/kubermatic
-COPY . .
+
+COPY ./cmd/user-ssh-keys-agent ./cmd/user-ssh-keys-agent
+COPY ./cmd/http-prober/api ./cmd/http-prober/api
+COPY ./pkg ./pkg
+COPY ./go.mod ./go.mod
+COPY ./go.sum ./go.sum
```

Comment:
Isn't one `COPY . .` much easier and future proof? Why do we copy everything individually?

Reply:
Since the entire docker context is quite big, I copied only the needed stuff to decrease the requested time. Copying everything is easier and future-proof, but also heavier, and since with this PR we want to decrease the time needed to build this image, I thought it may worth it.